From bca2e131f1dde6e70d1a77c3b1ebe3c7720eb4f8 Mon Sep 17 00:00:00 2001
From: Gioh Kim <gi-oh.kim@profitbricks.com>
Date: Tue, 31 Jan 2017 08:47:34 +0100
Subject: [832a404c8ce0] [PATCH 01/02] Task 08: add jiffies file

root@ws00837:/home/gohkim/study/eudyptula/task08# insmod drv.ko
root@ws00837:/home/gohkim/study/eudyptula/task08# ls /sys/kernel/debug/eudyptula/
id  jiffies
root@ws00837:/home/gohkim/study/eudyptula/task08# ls /sys/kernel/debug/eudyptula/jiffies
/sys/kernel/debug/eudyptula/jiffies
root@ws00837:/home/gohkim/study/eudyptula/task08# cat /sys/kernel/debug/eudyptula/jiffies
4298292650
root@ws00837:/home/gohkim/study/eudyptula/task08# cat /sys/kernel/debug/eudyptula/jiffies
4298293176
root@ws00837:/home/gohkim/study/eudyptula/task08# cat /sys/kernel/debug/eudyptula/jiffies
4298293344
root@ws00837:/home/gohkim/study/eudyptula/task08# cat /sys/kernel/debug/eudyptula/jiffies
4298294872
root@ws00837:/home/gohkim/study/eudyptula/task08# rmmod drv
root@ws00837:/home/gohkim/study/eudyptula/task08# ls /sys/kernel/debug/eudyptula/
ls: cannot access /sys/kernel/debug/eudyptula/: No such file or directory

Signed-off-by: Gioh Kim <gi-oh.kim@profitbricks.com>
---
 task08/drv.c | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git a/task08/drv.c b/task08/drv.c
index a4994f6..8cbb2e5 100644
--- a/task08/drv.c
+++ b/task08/drv.c
@@ -10,6 +10,7 @@
 #include <linux/uaccess.h>
 #include <linux/delay.h>
 #include <linux/debugfs.h>
+#include <linux/jiffies.h>
 
 MODULE_LICENSE("GPL");
 
@@ -47,7 +48,7 @@ static const struct file_operations eudyptula_id_fops = {
 
 static int __init hello_init(void)
 {
-	struct dentry *debugfs_id;
+	struct dentry *debugfs_id, *debugfs_jiffies;
 
 	eudyptula_debugfs_root = debugfs_create_dir("eudyptula", NULL);
 	if (!eudyptula_debugfs_root
@@ -69,6 +70,17 @@ static int __init hello_init(void)
 		pr_err("failed to create debugfs file: id\n");
 		return 0;
 	}
+
+	debugfs_jiffies = debugfs_create_u64("jiffies",
+					     S_IRUSR |
+					     S_IRGRP | S_IROTH,
+					     eudyptula_debugfs_root,
+					     &jiffies_64);
+	if (!debugfs_jiffies
+	    || debugfs_jiffies == ERR_PTR(-ENODEV)) {
+		pr_err("failed to create debugfs file: jiffies\n");
+		return 0;
+	}
 	return 0;
 }
 
-- 
2.5.0

